### DISCUSSION TITLE
How to deploy django with python-socketio in kubernetes?

### DISCUSSION THREAD

**Discussion by druuu** [miguelgrinberg/python-socketio#1037] [General]
Trying to run a django application with python-socketio.

Using below command for the running the pod: 
`exec gunicorn rc_app.wsgi:application  --name $APP_NAME --workers 1 --worker-class eventlet --worker-tmp-dir /dev/shm --bind 0.0.0.0:8000 --log-level debug
`

Running only one pod for now to test it out.

The pod is not processing request, response is conn refused.

**Comment by miguelgrinberg** [comment]
If your pod is not answering requests, then Socket.IO is not even running yet, so I don't see how the Socket.IO part of your application can be responsible.

I suggest you debug the interaction between Gunicorn and Kubernetes, which is where the problem is most likely.

**Reply by druuu** [reply]
@miguelgrinberg the app is running, I see the logs saying 'worker started with pid xx'.
The 'conn refused' log is from kubernetes readiness probe which i think is done via localhost network.
What I assume is that the socket io or the app is stuck and unable to respond to requests.

Do you see any issue in the command I posted?

**Reply by miguelgrinberg** [reply]
> the app is running, I see the logs saying 'worker started with pid xx'.

This means Gunicorn is running, not the WSGI app. You said above that your application does not respond to requests, so the app isn't running.

> Do you see any issue in the command I posted?

I have no way to know. Nothing is obviously wrong, but you could have made mistakes that are not obvious to me such as putting the app on the wrong port, for example. As I said, this is unlikely to be a Socket.IO problem, and at least requires more debugging on your part before Socket.IO can be considered part of the problem.

**Reply by ashwin31** [reply]
@miguelgrinberg can we communicate by email?

**Reply by druuu** [reply]
@miguelgrinberg do you provide paid service? We want to setup this app on kubernetes and require guidance from you. Let us know if you are interested.

**Reply by miguelgrinberg** [reply]
No, sorry. I can't really offer Kubernetes support, it's not an area of interest for me.

**Comment by bhakta0007** [comment]
If this is kubernetes, have you exposed a service and is there an endpoint for the service? This should be pretty straightforward to figure out. I am not using Django but I run flask and celery apps with socketio on k8

**Reply by ashwin31** [reply]
I need paid support if you can provide.

Next week.

**Reply by bhakta0007** [reply]
no - sorry - I have a 24x7 job already :- ) but if you can post a specific question - i'll be happy to answer
